### PR TITLE
add ini config to logstash callback

### PR DESCRIPTION
**Pull request by chris93111**
##### SUMMARY

Add ini configuration to callback logstash with new set_options

feature is present in 2.7

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

callback logstash

https://github.com/ansible/ansible/pull/66162

### CODE REVIEW

File: plugins/callback/logstash.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     short_description: Sends events to Logstash
     description:
       - This callback will report facts and task events to Logstash https://www.elastic.co/products/logstash
+    version_added: "2.3"
```
Comment (Akasurde) [inline plugins/callback/logstash.py:14]:
```suggestion

```
Not required.

File: plugins/callback/logstash.py [hunk 2]
```diff
@@ -68,41 +78,48 @@ class CallbackModule(CallbackBase):
...
+    This plugin makes use of the following environment variables or ini config:
         LOGSTASH_SERVER   (optional): defaults to localhost
         LOGSTASH_PORT     (optional): defaults to 5000
         LOGSTASH_TYPE     (optional): defaults to ansible
     """
 
     CALLBACK_VERSION = 2.0
     CALLBACK_TYPE = 'aggregate'
-    CALLBACK_NAME = 'community.general.logstash'
+    CALLBACK_NAME = 'logstash'
```
Comment (Akasurde) [inline plugins/callback/logstash.py:89]:
```suggestion
    CALLBACK_NAME = 'community.general.logstash'
```

File: plugins/callback/logstash.py [hunk 3]
```diff
@@ -19,16 +20,25 @@
         description: Address of the Logstash server
         env:
           - name: LOGSTASH_SERVER
+        ini:
+          - section: callback_logstash
+            key: server
```
Comment (felixfontein) [inline plugins/callback/logstash.py:25]:
```suggestion
            key: server
            version_added: 1.0.0
```
Same for the other new options.

File: plugins/callback/logstash.py (hunk 1 above)
Comment (chris93111) [reply plugins/callback/logstash.py]: ok

File: plugins/callback/logstash.py (hunk 2 above)
Comment (chris93111) [reply plugins/callback/logstash.py]: ok

File: plugins/callback/logstash.py (hunk 3 above)
Comment (chris93111) [reply plugins/callback/logstash.py:24]: @felixfontein version_added is needed to all key in section ini ?

Comment (felixfontein) [reply plugins/callback/logstash.py:24]: Only for the ones you added (which in this case seems to be all of them).

**Comment by felixfontein** [comment]
You also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to) (`minor_changes:`).

**Comment by chris93111** [comment]
> You also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to) (`minor_changes:`).

ok is added

File: changelogs/fragments/610_logstash_callback_add_ini_config.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- "add ini config to logstash callback"
```
Comment (felixfontein) [inline changelogs/fragments/610_logstash_callback_add_ini_config.yml:2]:
```suggestion
- "logstash callback - add ini config (https://github.com/ansible-collections/community.general/pull/610)."
```

Comment (chris93111) [reply changelogs/fragments/610_logstash_callback_add_ini_config.yml]: thanks :)

File: plugins/callback/logstash.py [hunk 5]
```diff
@@ -19,16 +19,26 @@
...
+            version_added: 1.0.0
         default: localhost
       port:
         description: Port on which logstash is listening
         env:
             - name: LOGSTASH_PORT
+        ini:
+          - section: callback_logstash
+            key: port
```
Comment (felixfontein) [inline plugins/callback/logstash.py:33]: You also need a `version_added` here, and also for the third ini entry.

Comment (chris93111) [reply plugins/callback/logstash.py:33]: ok

**Comment by felixfontein** [comment]
@chris93111 thanks for this contribution!
@Akasurde thanks for reviewing!

**Pull request merged**